Code/GraphMol/FMCS/MaximumCommonSubgraph.cpp

```diff
@@ -68,50 +70,46 @@ void MaximumCommonSubgraph::init() {
 #ifdef DUP_SUBSTRUCT_CACHE
   DuplicateCache.clear();
 #endif
-  void* userData = Parameters.CompareFunctionsUserData;
 
-  size_t nq = 0;
 #ifdef FAST_SUBSTRUCT_CACHE
-  // fill out RingMatchTables to check cache Hash collision by checking match a
-  // part of Query to Query
-  if (!userData  // predefined functor - compute RingMatchTable for all targets
-      && (Parameters.BondCompareParameters.CompleteRingsOnly ||
-          Parameters.BondCompareParameters.RingMatchesRingOnly ||
-          Parameters.AtomCompareParameters.RingMatchesRingOnly)) {
-    RingMatchTables.init(QueryMolecule);
-    Parameters.CompareFunctionsUserData = &RingMatchTables;
-    RingMatchTables.computeRingMatchTable(QueryMolecule, QueryMolecule,
-                                          Parameters);
+  loadMatchTables(QueryMolecule, QueryAtomMatchTable, QueryBondMatchTable);
```

Comment:
This is my most aggressive refactoring and I can undo it. I thought it was an opportunity to clarify the code and to remove duplicate code where we load the match tables for QueryMolecule and the targets.